ChronoKeeper/StoryPipeline.cpp

```diff
@@ -1,44 +1,271 @@
+	if( !result.second)
+	{ 
+	    return storyTimelineMap.end(); 
+	}	
+        else
+	{
+	    timelineStart -= chunkGranularity;
+	    return result.first;
+	}
+}
+/////////////////////////////
+
+std::map<uint64_t, chronolog::StoryChunk>::iterator chronolog::StoryPipeline::appendStoryChunk()
+{
+  // append the next storyChunk at the end of storyTimeline and return the iterator to the new node
+#ifdef TRACE_CHUNKING
+    	std::chrono::time_point<std::chrono::system_clock,std::chrono::nanoseconds> epoch_time_point{};
+    	auto chunk_start_point = epoch_time_point + std::chrono::nanoseconds(timelineEnd);
+    	std::time_t time_t_chunk_start = std::chrono::high_resolution_clock::to_time_t(chunk_start_point);
+    	auto chunk_end_point = epoch_time_point + std::chrono::nanoseconds(timelineEnd+chunkGranularity);
+    	std::time_t time_t_chunk_end = std::chrono::high_resolution_clock::to_time_t(chunk_end_point); 
+    	std::cout <<"StoryPipeline: storyId { "<< storyId<<"} append chunk {" << timelineEnd <<" "<< std::ctime(&time_t_chunk_start) 
+		<<"} {" << timelineEnd+chunkGranularity <<" " <<std::ctime(&time_t_chunk_end) <<"}"<<std::endl;
+#endif 
+	auto result= storyTimelineMap.insert( std::pair<uint64_t, chronolog::StoryChunk>( timelineEnd, chronolog::StoryChunk(storyId, timelineEnd, timelineEnd+chunkGranularity)));
+	if( !result.second)
+	{ 
+	    return storyTimelineMap.end(); 
+	}	
+        else
+	{
+	    timelineEnd += chunkGranularity;
+	    return result.first;
+	}
+}
+//////////////////////
+
+void chronolog::StoryPipeline::collectIngestedEvents()
+{
+	activeIngestionHandle->swapActiveDeque(); 
+
+	if( !activeIngestionHandle->getPassiveDeque().empty())
+	{   mergeEvents(activeIngestionHandle->getPassiveDeque()); }
 
-return status;
 }
 ////////////////////
 
-int chronolog::StoryPipeline::mergeEvents(chronolog::StoryChunk const&)
+void chronolog::StoryPipeline::mergeEvents(chronolog::EventDeque & event_deque)
+{
+
+	if(event_deque.empty())
+	{ return;}
+
+	std::lock_guard<std::mutex> lock(sequencingMutex);
+	chl::LogEvent event;
+	// the last chunk is most likely the one that would get the events, so we'd start with the last 
+	// chunk and do the lookup only if it's not the one
+	// NOTE: we should never have less than 2 chunks in the active storyTimelineMap !!!
+    	std::map<uint64_t, chronolog::StoryChunk>::iterator chunk_to_merge_iter = --storyTimelineMap.end();
+	while( !event_deque.empty())
+	{
+		event = event_deque.front();
+		{std::cout << "StoryPipeline: {"<< storyId<<":"<<timelineStart<<":"<<timelineEnd<<"} merging event  {"<< event.time()<<"}"<<std::endl; }
+        	if( timelineStart <= event.time() && event.time() < timelineEnd)
+        	{       // we expect the events in the deque to be mostly monotonous
+			// so we'd try the most recently used chunk first and only look for the new chunk 
+			// if the event does not belong to the recently used chunk
+        		if( !(*chunk_to_merge_iter).second.insertEvent(event) )
```

Comment:
Having a dedicated function to detect if an event falls into a StoryChunk is more intuitive instead of calling insert and expecting a return value of 0.

Reply:
Take a look at StoryChunk.h line 60 :       the first thing that insertEvent does is checks that event.time() falls between chunk startTime and endTime.
While I can add a dedicated function to do this check it seems redundant in the case when what we want is not kjust do the comparison check but actually insert event into the StoryChunk if it belongs